selftests/unit/test_utils_asset.py

```diff
@@ -18,13 +18,15 @@ def setUp(self):
             f.write('Test!')
         self.url = 'file://%s' % self.localpath
         self.cache_dir = tempfile.mkdtemp(dir=self.basedir)
+        self.expire = 0
```

Comment:
You could add expiration unittest. You can check for the tmpdir with a negative expiration time :wink: